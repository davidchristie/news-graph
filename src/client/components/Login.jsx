import React, { PropTypes } from 'react'

export default function Login (props) {
  const onSubmit = props.onSubmit
  return (
    <div className='dropdown'>
      <button className='btn btn-success dropdown-toggle' type='button' id='dropdownLogin' data-toggle='dropdown' aria-haspopup='true' aria-expanded='false'>
        Login
      </button>
      <div className='dropdown-menu dropdown-menu-right' aria-labelledby='dropdownLogin' style={{minWidth: '300px'}}>

        <form role='form' acceptCharset='UTF-8' id='login-nav' onSubmit={onSubmit}>

          <div className='dropdown-item'>
            <label className='sr-only' htmlFor='email'>Email address</label>
            <input type='email' className='form-control' id='email' name='email' placeholder='Email address' required />
          </div>

          <div className='dropdown-item'>
            <label className='sr-only' htmlFor='password'>Password</label>
            <input type='password' className='form-control' id='password' name='password' placeholder='Password' required />
          </div>

          <div className='dropdown-item'>
            <button type='submit' className='btn btn-outline-success btn-block'>Sign in</button>
          </div>

        </form>

      </div>
    </div>
  )
}

Login.propTypes = {
  onSubmit: PropTypes.func.isRequired
}
