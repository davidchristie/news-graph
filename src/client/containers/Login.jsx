import React from 'react'
import { connect } from 'react-redux'

import { login } from '../actions/account'
import Alerts from '../components/Alerts'

export class Login extends React.Component {

  constructor (props) {
    super(props)
    this.state = {alerts: []}
  }

  render () {
    const onSubmit = event => {
      event.preventDefault()
      const form = event.target
      const inputs = form.getElementsByTagName('input')
      const email = inputs[0].value
      const password = inputs[1].value
      this.props.dispatch(login({email, password}))
        .then(({ alerts, success }) => {
          if (!success) {
            this.setState({alerts})
          }
        })
    }
    return (
      <div className='dropdown'>
        <button className='btn btn-success dropdown-toggle' type='button' id='dropdownLogin' data-toggle='dropdown' aria-haspopup='true' aria-expanded='false'>
          Login
        </button>
        <div className='dropdown-menu dropdown-menu-right' aria-labelledby='dropdownLogin' style={{minWidth: '300px'}}>

          <Alerts alerts={this.state.alerts} />

          <form role='form' acceptCharset='UTF-8' id='login-nav' onSubmit={onSubmit}>

            <div className='dropdown-item'>
              <label className='sr-only' htmlFor='email'>Email address</label>
              <input type='email' className='form-control' id='email' name='email' placeholder='Email address' required />
            </div>

            <div className='dropdown-item'>
              <label className='sr-only' htmlFor='password'>Password</label>
              <input type='password' className='form-control' id='password' name='password' placeholder='Password' required />
            </div>

            <div className='dropdown-item'>
              <button type='submit' className='btn btn-outline-success btn-block'>Sign in</button>
            </div>

          </form>

        </div>
      </div>
    )
  }
}

export default connect()(Login)
