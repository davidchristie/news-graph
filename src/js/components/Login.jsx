import React from 'react'

export default class Layout extends React.Component {
  render() {
    return (
      <div className="dropdown">
        <button className="btn btn-success dropdown-toggle" type="button" id="dropdownLogin" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
          Login
        </button>
        <div className="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownLogin" style={{minWidth: '300px'}}>

          <form role="form" method="post" action="/users/login" acceptCharset="UTF-8" id="login-nav">

            <div className="dropdown-item">
               <label className="sr-only" htmlFor="email">Email address</label>
               <input type="email" className="form-control" id="email" name="email" placeholder="Email address" required />
            </div>

            <div className="dropdown-item">
               <label className="sr-only" htmlFor="password">Password</label>
               <input type="password" className="form-control" id="password" name="password" placeholder="Password" required />
            </div>

            <div className="dropdown-item">
               <button type="submit" className="btn btn-outline-success btn-block">Sign in</button>
            </div>

          </form>

        </div>
      </div>
    )
  }
}

// <div className="dropdown">
//   <a href="#" className="dropdown-toggle" data-toggle="dropdown">
//     <b>Login</b> <span className="caret"></span>
//   </a>
//   <ul className="dropdown-menu">
//     <li>
//         <div className="col-xs-12">
//
//            <form className="form" role="form" method="post" action="/users/login" acceptCharset="UTF-8" id="login-nav">
//
//               <div className="form-group">
//                  <label className="sr-only" htmlFor="email">Email address</label>
//                  <input type="email" className="form-control" id="email" name="email" placeholder="Email address" required />
//               </div>
//
//               <div className="form-group">
//                  <label className="sr-only" htmlFor="password">Password</label>
//                  <input type="password" className="form-control" id="password" name="password" placeholder="Password" required />
//               </div>
//
//               <div className="form-group">
//                  <button type="submit" className="btn btn-default btn-block">Sign in</button>
//               </div>
//
//            </form>
//         </div>
//     </li>
//   </ul>
// </div>
