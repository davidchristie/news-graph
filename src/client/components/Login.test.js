/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Login from './Login'

const onSubmit = () => {}

test('Login renders correctly', () => {
  const wrapper = shallow(<Login onSubmit={onSubmit} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
