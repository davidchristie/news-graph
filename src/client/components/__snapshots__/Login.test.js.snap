exports[`test Login renders correctly 1`] = `
<div
  className="dropdown">
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="btn btn-success dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownLogin"
    type="button">
    Login
  </button>
  <div
    aria-labelledby="dropdownLogin"
    className="dropdown-menu dropdown-menu-right"
    style={
      Object {
        "minWidth": "300px",
      }
    }>
    <form
      acceptCharset="UTF-8"
      id="login-nav"
      onSubmit={[Function]}
      role="form">
      <div
        className="dropdown-item">
        <label
          className="sr-only"
          htmlFor="email">
          Email address
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          placeholder="Email address"
          required={true}
          type="email" />
      </div>
      <div
        className="dropdown-item">
        <label
          className="sr-only"
          htmlFor="password">
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          placeholder="Password"
          required={true}
          type="password" />
      </div>
      <div
        className="dropdown-item">
        <button
          className="btn btn-outline-success btn-block"
          type="submit">
          Sign in
        </button>
      </div>
    </form>
  </div>
</div>
`;
